rrule/src/parser/datetime.rs

```diff
@@ -0,0 +1,262 @@
+        })?;
+
+    // Apply timezone appended to the datetime before converting to UTC.
+    // For more info https://icalendar.org/iCalendar-RFC-5545/3-3-5-date-time.html
+    let datetime: chrono::DateTime<chrono::Utc> = if flags.zulu_timezone_set {
+        // If a `Z` is present, UTC should be used.
+        chrono::DateTime::<_>::from_utc(datetime, chrono::Utc)
+    } else {
+        // If no `Z` is present, local time should be used.
+        use chrono::offset::LocalResult;
+        // Get datetime in local time or machine local time.
+        // So this also takes into account daylight or standard time (summer/winter).
+        match tz {
+            Some(tz) => {
+                // Use the timezone specified in the `tz`
+                match tz.from_local_datetime(&datetime) {
+                    LocalResult::None => Err(ParseError::InvalidDateTimeInLocalTimezone {
+                        value: dt.into(),
+                        property: property.into(),
+                    }),
+                    LocalResult::Single(date) => Ok(date),
+                    LocalResult::Ambiguous(date1, date2) => {
+                        Err(ParseError::DateTimeInLocalTimezoneIsAmbiguous {
+                            value: dt.into(),
+                            property: property.into(),
+                            date1: date1.to_rfc3339(),
+                            date2: date2.to_rfc3339(),
+                        })
+                    }
+                }?
+                .with_timezone(&chrono::Utc)
+            }
+            None => {
+                // Use current system timezone
+                // TODO Add option to always use UTC when this is executed on a server.
+                let local = chrono::Local;
+                match local.from_local_datetime(&datetime) {
+                    LocalResult::None => Err(ParseError::InvalidDateTimeInLocalTimezone {
+                        value: dt.into(),
+                        property: property.into(),
+                    }),
+                    LocalResult::Single(date) => Ok(date),
+                    LocalResult::Ambiguous(date1, date2) => {
+                        Err(ParseError::DateTimeInLocalTimezoneIsAmbiguous {
+                            value: dt.into(),
+                            property: property.into(),
+                            date1: date1.to_rfc3339(),
+                            date2: date2.to_rfc3339(),
+                        })
+                    }
+                }?
+                .with_timezone(&chrono::Utc)
+            }
+        }
+    };
+
+    // Apply timezone from `TZID=` part (if any), else set datetime as UTC
+    let datetime_with_timezone = datetime.with_timezone(&tz.unwrap_or(UTC));
+
+    Ok(datetime_with_timezone)
+}
+
+/// Attempts to convert a `str` to a `Weekday`.
+pub(crate) fn str_to_weekday(d: &str) -> Result<Weekday, ParseError> {
+    let day = match &d.to_uppercase()[..] {
+        "MO" => Weekday::Mon,
```

Comment:
Not your changes, but I think it's better to define ALLL static ASCII strings as binary `b"MO"`. And instead of `&str`, pass `&[u8]` around.
Sure it should be another PR, it's a big and important change!

Reply:
Hmm, that is interesting and definitely something we can look into in another PR!